barretenberg/cpp/src/barretenberg/stdlib/recursion/honk/verifier/protogalaxy_recursive_verifier.test.cpp

```diff
@@ -299,82 +294,69 @@ template <typename RecursiveFlavor> class ProtoGalaxyRecursiveTests : public tes
+        info("Decider Recursive Verifier: num gates = ", decider_circuit.num_gates);
+
+        {
+            auto composer = Composer();
+            auto instance = composer.create_prover_instance(decider_circuit);
+            auto verification_key = composer.compute_verification_key(instance);
+            auto prover = composer.create_prover(instance);
+            auto verifier = composer.create_verifier(verification_key);
+            auto proof = prover.construct_proof();
+            bool verified = verifier.verify_proof(proof);
 
-        // We expect the decider circuit check to fail due to the bad proof
-        EXPECT_FALSE(outer_decider_circuit.check_circuit());
+            EXPECT_FALSE(verified);
+        }
     };
 
     static void test_tampered_accumulator()
     {
-        // Create two arbitrary circuits for the first round of folding
-        InnerBuilder builder1;
-
-        create_inner_circuit(builder1);
-        InnerBuilder builder2;
-        builder2.add_public_variable(FF(1));
-        create_inner_circuit(builder2);
-
-        InnerComposer inner_composer = InnerComposer();
-        auto instance1 = inner_composer.create_instance(builder1);
-        auto instance2 = inner_composer.create_instance(builder2);
-        auto instances = std::vector<std::shared_ptr<Instance>>{ instance1, instance2 };
 
-        auto accumulator = fold_and_verify_native(instances, inner_composer);
+        auto composer = Composer();
+        auto [prover_accumulator, verifier_accumulator] = fold_and_verify_native(composer);
 
         // Create another circuit to do a second round of folding
-        InnerBuilder builder3;
-        create_inner_circuit(builder3);
-        auto instance3 = inner_composer.create_instance(builder3);
+        Builder builder3;
+        create_function_circuit(builder3);
+        auto prover_inst = composer.create_prover_instance(builder3);
+        auto verifier_inst = composer.create_verifier_instance(prover_inst);
 
-        // Tamper with the accumulator
-        instances = std::vector<std::shared_ptr<Instance>>{ accumulator, instance3 };
-        accumulator->prover_polynomials.w_l[1] = FF::random_element();
+        prover_accumulator->prover_polynomials.w_l[1] = FF::random_element();
 
         // Generate a folding proof
-        auto inner_folding_prover = inner_composer.create_folding_prover(instances);
-        auto inner_folding_proof = inner_folding_prover.fold_instances();
+        auto folding_prover = composer.create_folding_prover({ prover_accumulator, prover_inst });
+        auto folding_proof = folding_prover.fold_instances();
 
         // Create a recursive folding verifier circuit for the folding proof of the two instances
-        OuterBuilder outer_folding_circuit;
-        FoldingRecursiveVerifier verifier{ &outer_folding_circuit };
-        verifier.verify_folding_proof(inner_folding_proof.folding_data);
-        EXPECT_EQ(outer_folding_circuit.check_circuit(), false);
```

Comment:
Recall prover used to send \phi, target sum and \vec{\beta} to the verifier and the verifier would validate against what it has computed (achieved with `assert_equals` inside the circuit) so for different target sums because of tampering `check_circuit` failed